frontend/src/search/search.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { PropTypes } from 'react'
+
+import './search.css';
+
+export default class Search extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      search_value: '',
+      search_by_value: 'title',
+      sort_by: 'release_date',
+    };
+
+    this.searchMovies = this.searchMovies.bind(this);
+    this.searchByTitle = this.searchByTitle.bind(this);
+    this.searchByGender = this.searchByGender.bind(this);
+    this.sortByDate = this.sortByDate.bind(this);
+    this.sortByRating = this.sortByRating.bind(this);
+    this.newSearchSubmitted = this.newSearchSubmitted.bind(this);
+  }
+
+  searchMovies(evt) {
+    evt.persist();
+    this.setState(state => ({ search_value: evt.target.value }));
+  }
+
+  newSearchSubmitted() {
+    this.props.applyNewSearch(this.state.search_value, this.state.search_by_value, this.state.sort_by);
+  }
+
+  searchByTitle(evt) {
+    console.log(evt.target.value);
+    evt.persist();
+    this.setState({ search_by_value: evt.target.value });
+  }
+
+  searchByGender(evt) {
+    console.log(evt.target.value);
+    evt.persist();
+    this.setState({ search_by_value: evt.target.value });
+  }
+
+  sortByDate(evt) {
+    console.log(evt.target.value);
+    evt.persist();
+    this.setState({ sort_by: evt.target.value });
+  }
+
+  sortByRating(evt) {
+    console.log(evt.target.value);
+    evt.persist();
+    this.setState({ sort_by: evt.target.value });
+  }
```

Comment:
Methods are ABSOLUTELY the same inside, the only name is different. Can we avoid copy-pasting?